### ISSUE TITLE
An option to make Windhawk to use the host system window theme/style?

### ISSUE THREAD

**Issue by Anixx** [ramensoftware/windhawk#62]
It would be very useful, especially the window decorations/frame because currently it is just filled grey rectangle, if you have another app with dark grey color on the background (such as Discord), the window border is simply invisible.

**Comment by m417z** [comment]
It's not currently supported, but since the Windhawk UI is based on VSCode, you can choose another theme:
* Press Ctrl+Shift+Alt+P (just Ctrl+P if in editing mode)
* Type and select `Preferences: Color Theme`
* Choose another theme

Note that the Windhawk UI doesn't look so good with some themes, especially light themes.

![image](https://user-images.githubusercontent.com/4129781/229271400-d61af338-1c86-4502-b722-5fac8a674ab1.png)

**Comment by Anixx** [comment]
I wonder what is this new fashion to make windows without borders?
How do they work with it themselves?
![изображение](https://user-images.githubusercontent.com/1271736/232338453-45616db3-366b-40bc-8f0a-a299b4e2fce3.png)
Am I correct that Microsoft decided to use a browser-based toolkit from Google for their Visual Studio after they decided to switch to the browser from their main competitor? Aren't they crazy?

**Comment by m417z** [comment]
There's a border in Windows 11.

![image](https://user-images.githubusercontent.com/4129781/232338684-1f5bd223-9f91-456e-a462-e6234ce5a948.png)

I found these options, you might want to try them: https://github.com/microsoft/vscode/pull/82781#issuecomment-543221612
Instructions:
* Press Ctrl+Alt+Shift+P and select `Preferences: Open User Settings (JSON)`
* Add a new entry like this:
```json
    "workbench.colorCustomizations": {
        "window.activeBorder": "#00ff00",
        "window.inactiveBorder": "#ff0000"
    }
```
* Configure the colors as desired
* Restart Windhawk

Not sure that it will work, it didn't for me, but maybe that's because it's not supported on Windows 11.

> Am I correct that Microsoft decided to use a browser-based toolkit from Google

Electron is based on Chromium, but it was originally developed by GitHub for Atom, and now GitHub is part of Microsoft. And historically Chromium is based on WebKit, so not 100% of a Google's creation either way.

> for their Visual Studio

Visual Studio Code is an unrelated project, like Java is unrelated to JavaScript. And yes, Microsoft is known to be bad with names.

> after they decided to switch to the browser from their main competitor? Aren't they crazy?

Would it be better for them to stay with the IE-based Edge? Probably not in the state that it was, and improving it was probably deemed to be too much of an investment. On the one hand, it's bad for competition, but on the other hand, it dilutes Google's control over Chromium, which is IMO a good thing.

**Comment by Anixx** [comment]
> Not sure that it will work, it didn't for me, but maybe that's because it's not supported on Windows 11.

Did not work for me either...

> Would it be better for them to stay with the IE-based Edge? Probably not in the state that it was, and improving it was probably deemed to be too much of an investment. On the one hand, it's bad for competition, but on the other hand, it dilutes Google's control over Chromium, which is IMO a good thing.

Microsoft did a lot of stupid things since Bill Gates left the company. One of them is ditching VBA. Then, migrating to JavaScript-based web design. 
And moving to Chrome-based browser... When I first saw Chrome ads on Google's site that advertised it as the fastest... I thought "who will buy this?"... 

And it seems the folks from Microsoft were the ones. They migrated to a browser which became completely detached from their OS. In IE era mshtlm was a standard part of the system. Anyone could embed up-to-date browser control in their applications. Basically, I could make a new browser of my liking in VB6 in 10 minutes. For instance, with two vertical bookmarks bars on right and left and whatever.
Now, what? I have to ship a browser engine with my program if I want it to display web content?
Besides this, Chrome is sumply an awful browser, it does not, for instance, support MathML, which is the main reason, why Wikipedia still uses SVG for formulas rendering. Its font rendering is also far inferiot than the Windows built-in cleartype (and anyway, why an OS shoud include separate fonts engines for different purposes?). Chrome is simply not integrated with Windows.

On the other hand, Chrome is of course better than the Metro-based Edge, it is a desktop app after all.

**Comment by m417z** [comment]
> Did not work for me either...

Yes, you're right, turns out that it was removed in a recent version: https://github.com/microsoft/vscode/issues/160159
We'll have to wait for Microsoft to implement https://github.com/microsoft/vscode/issues/153222

> Now, what? I have to include a browser engine with my program if I want it to display web content?

There's the WebView2 control. For example, Tauri, a lightweight Electron alternative, uses it. There was an issue about making VSCode more lightweight with it but it was rejected: https://github.com/microsoft/vscode/issues/118308

**Comment by SirWobbyTheFirst** [comment]
So I was able to set it to use the native theme by doing the Ctrl+Alt+Shift+P option as mentioned and then typing preferences, there was then an option to open the Settings UI and then I searched for title in the settings page and was able to switch it to using the Native option, WH prompted to restart and applied the theme.

It shows the Menubar by default but going into Settings again and searching for menu has an option to hide the menubar. The only issue is that you cannot exit the Settings page, so you have to close and restart the app, but that's a minor hiccup.

![image](https://github.com/ramensoftware/windhawk/assets/47334951/944ed001-177b-4525-a482-403296dc12d4)

**Comment by Anixx** [comment]
In Classic Theme the native window border is glitchy, not getting redrawn when needed.

But if after enabling classic theming, add it to Classic Theme Browser Fix mod, the border becomes Aero Lite:

![изображение](https://github.com/ramensoftware/windhawk/assets/1271736/2c08a9d3-5d69-4cce-8c6f-33b210d8b1f6)

**Comment by YellowAfterlife** [comment]
I was similarly pleased to find that I can change the code editor theme and then slightly less pleased to find that it applies to the background of the main window (couldn't care less that the cards are still white-on-black, I don't have to look at them for too long).

If you update to a newer Electron version later on (v24 or newer, [by the looks of it?](https://www.electronjs.org/docs/latest/tutorial/electron-timelines)), you could change your
```css
color: hsla(0,0%,100%,.85);
```
to
```css
color: color-mix(in srgb, var(--vscode-editor-foreground), transparent 15%);
```
and that would respect the theme's font color.

Or could just pick a similar color from the theme unless the slightly-muted white is all too precious for aesthetics.

For the time I settled for [someone's plugin](https://github.com/Clarkkkk/vscode-auto-light-dark-theme) that adds a light/dark theme toggle button to the status bar.

**Comment by m417z** [comment]
A light theme is now available in Windhawk 2.0 alpha which can be downloaded [here](https://github.com/ramensoftware/windhawk/releases).
